src/sharding/ShardingManager.js

```diff
@@ -56,58 +57,109 @@ class ShardingManager extends EventEmitter {
       if (this.totalShards !== Math.floor(this.totalShards)) {
         throw new RangeError('CLIENT_INVALID_OPTION', 'Amount of shards', 'an integer.');
       }
+    } else {
+      this.autoShards = true;
     }
 
     /**
-     * Whether shards should automatically respawn upon exiting
-     * @type {boolean}
-     */
+    * Whether shards should automatically respawn upon exiting
+    * @type {boolean}
+    */
     this.respawn = options.respawn;
 
     /**
-     * An array of arguments to pass to shards
-     * @type {string[]}
-     */
+    * An array of arguments to pass to shards
+    * @type {string[]}
+    */
     this.shardArgs = options.shardArgs;
 
     /**
-     * Token to use for obtaining the automatic shard count, and passing to shards
-     * @type {?string}
-     */
+    * Token to use for obtaining the automatic shard count, and passing to shards
+    * @type {?string}
+    */
     this.token = options.token ? options.token.replace(/^Bot\s*/i, '') : null;
 
     /**
-     * A collection of shards that this manager has spawned
-     * @type {Collection<number, Shard>}
-     */
+    * A collection of shards that this manager has spawned
+    * @type {Collection<number, Shard>}
+    */
     this.shards = new Collection();
   }
 
   /**
-   * Spawns a single shard.
-   * @param {number} [id=this.shards.size] ID of the shard to spawn -
-   * **This is usually not necessary to manually specify.**
-   * @returns {Shard}
-   */
+  * Spawns a single shard.
+  * @param {number} [id=this.shards.size] ID of the shard to spawn -
+  * **This is usually not necessary to manually specify.**
+  * @returns {Shard}
+  */
   createShard(id = this.shards.size) {
     const shard = new Shard(this, id, this.shardArgs);
     this.shards.set(id, shard);
     /**
-     * Emitted upon creating a shard.
-     * @event ShardingManager#shardCreate
-     * @param {Shard} shard Shard that was created
-     */
+    * Emitted upon creating a shard.
+    * @event ShardingManager#shardCreate
+    * @param {Shard} shard Shard that was created
+    */
     this.emit('shardCreate', shard);
     return shard;
   }
+  /**
+  * Kills all Shards without them Respawning.
+  */
+  killAll() {
+    const chk = false;
+    if (this.respawn) {
+      this.respawn = false;
+      chk = true;
+    }
+    const promises = [];
+    for (const shard of this.shards.values()) promises.push(shard.kill());
+    await Promise.all(promises);
+    if (chk) {
+      this.respawn = true;
+    }
+  }
+  /**
+  * Checks the Shard Amount, then Respawns the Correct Amount of Shards.
```

Comment:
Too much of `This Capitalization`, Should be more lowercase

Reply:
Please tell me if that is the way you wanted. I changed them. Sorry way I type is weird.